src/ncp/ncp_base_radio.cpp

```diff
@@ -421,6 +429,50 @@ otError NcpBase::HandlePropertySet_SPINEL_PROP_STREAM_RAW(uint8_t aHeader)
     return error;
 }
 
+template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_RCP_MAC_KEY>(void)
+{
+    otError        error = OT_ERROR_NONE;
+    uint8_t        keyIdMode;
+    uint16_t       keySize;
+    const uint8_t *prevKey;
+    const uint8_t *currKey;
+    const uint8_t *nextKey;
+
+    SuccessOrExit(error = mDecoder.ReadUint8(keyIdMode));
+    VerifyOrExit(keyIdMode == Mac::Frame::kKeyIdMode1, error = OT_ERROR_INVALID_ARGS);
```

Comment:
Since `keyIdMode` could only be `1`, shall we remove the parameter from the functions and always let `keyIdMode` to be `1` on RCP? Or it is possible to extend further in the future?

Reply:
I think it is better to have the ability to extend this property in the future although only one type of key ID mode is used right now.